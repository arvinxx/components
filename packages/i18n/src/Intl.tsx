import React from 'react';
import { IntlProvider as IntlProvider_ } from 'react-intl';
import type { IntlProviderProps } from './type';

export const IntlProvider: IntlProviderProps = (props) => (
  <IntlProvider_ {...props} />
);
