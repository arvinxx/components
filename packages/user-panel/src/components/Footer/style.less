.avx-user-panel-footer {
  &-divider {
    margin: 8px 0;
    padding: 0 16px;
    color: hsla(0, 0%, 0%, 0.25) !important;
    font-size: 14px !important;
  }

  &-other {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 24px;
    font-size: 12px;
    line-height: 22px;
  }

  &-register {
    margin-top: 8px;
    font-size: 16px;
  }
}
