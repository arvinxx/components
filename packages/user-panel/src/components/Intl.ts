import { useTranslation } from 'react-i18next';

export const useFormatMessage = () => {
  const { t } = useTranslation();

  return (id, values?) => t(id, values);
};
