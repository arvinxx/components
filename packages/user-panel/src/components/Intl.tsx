import React from 'react';
import { IntlProvider as IntlProvider_ } from 'react-intl';
import type { LocaleMessages } from '../type';

export const IntlProvider: React.FC<
  Omit<React.ComponentProps<typeof IntlProvider_>, 'messages'> & {
    messages: LocaleMessages;
  }
> = (props) => <IntlProvider_ {...props} />;
