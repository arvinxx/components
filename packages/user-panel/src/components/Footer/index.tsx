import React from 'react';
import type { FC } from 'react';
import type { NavigateFunction } from 'react-router';
import { useNavigate } from 'react-router';

import { Typography, Divider } from 'antd';

import WechatLogin from '../WechatLogin';

import { useFormatMessage } from '../Intl';
import type { PanelContentType } from '../../types';

import './style.less';

const { Text } = Typography;

export interface FooterProps {
  /**
   * 类型
   */
  type: PanelContentType;
  /**
   * @title 微信登录方法
   */
  onWechatLoginClick?: () => void;

  /**
   * @title 注册页面 Url
   */
  onRegisterClick?: (navigate: NavigateFunction) => void;
}

export const Footer: FC<FooterProps> = ({
  type,
  onWechatLoginClick,
  onRegisterClick,
}) => {
  const f = useFormatMessage();

  const navigate = useNavigate();

  const onClick = () => {
    switch (type) {
      case 'login':
        onRegisterClick?.(navigate);
        break;
      case 'register':
        break;
      case 'forgot':
        break;
      default:
        return '';
    }
  };
  return (
    <div className="avx-user-panel-footer">
      <Divider dashed className="avx-user-panel-footer-divider">
        {f('login.or')}
      </Divider>
      {onWechatLoginClick ? <WechatLogin login={onWechatLoginClick} /> : null}
      <div className="avx-user-panel-footer-other">
        <Text>
          {f(type === 'login' ? 'login.no-account' : 'register.have-account')}
        </Text>
        <div className="avx-user-panel-footer-register">
          <a onClick={() => onClick()}>
            {f(type === 'login' ? 'login.signup' : 'register.sign-in')}
          </a>
        </div>
      </div>
    </div>
  );
};
