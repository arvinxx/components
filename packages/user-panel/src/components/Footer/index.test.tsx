import React from 'react';
import { render } from '@testing-library/react';

import Footer from './index';
import { IntlProvider } from '../Intl';

describe('Footer 组件', () => {
  it('登录页脚', () => {
    const { container } = render(
      <IntlProvider>
        <Footer type={'login'} />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });
  it('微信登陆页面', () => {
    const { container } = render(
      <IntlProvider>
        <Footer type={'register'} wechatLogin={() => {}} />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });
  it('忘记密码', () => {
    const { container } = render(
      <IntlProvider>
        <Footer type={'forgot'} />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });
  it('空', () => {
    const { container } = render(
      <IntlProvider>
        <Footer
          // @ts-ignore
          type={''}
        />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
