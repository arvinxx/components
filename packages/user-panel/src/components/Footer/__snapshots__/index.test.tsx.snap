// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer 组件 登录页脚 1`] = `
<div>
  <div
    class="avx-user-panel-footer"
  >
    <div
      class="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center ant-divider-dashed avx-user-panel-footer-divider"
      role="separator"
    >
      <span
        class="ant-divider-inner-text"
      >
        或
      </span>
    </div>
    <div
      class="avx-user-panel-footer-other"
    >
      <span
        class="ant-typography"
      >
        没有账号？
      </span>
      <div
        class="avx-user-panel-footer-register"
      >
        <a>
          创建账户
        </a>
      </div>
    </div>
  </div>
</div>
`;
